app/src/main/java/org/schabi/newpipe/settings/SelectPlaylistFragment.java

```diff
@@ -0,0 +1,226 @@
+package org.schabi.newpipe.settings;
+
+import android.app.Activity;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.nostra13.universalimageloader.core.DisplayImageOptions;
+import com.nostra13.universalimageloader.core.ImageLoader;
+
+import org.schabi.newpipe.NewPipeDatabase;
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.database.AppDatabase;
+import org.schabi.newpipe.database.LocalItem;
+import org.schabi.newpipe.database.playlist.PlaylistLocalItem;
+import org.schabi.newpipe.database.playlist.PlaylistMetadataEntry;
+import org.schabi.newpipe.database.playlist.model.PlaylistRemoteEntity;
+import org.schabi.newpipe.local.playlist.LocalPlaylistManager;
+import org.schabi.newpipe.local.playlist.RemotePlaylistManager;
+import org.schabi.newpipe.report.ErrorActivity;
+import org.schabi.newpipe.report.UserAction;
+import org.schabi.newpipe.util.PlaylistItemsUtils;
+
+import java.util.List;
+import java.util.Vector;
+
+import io.reactivex.Flowable;
+import io.reactivex.disposables.Disposable;
+
+public class SelectPlaylistFragment extends DialogFragment {
+    /**
+     * This contains the base display options for images.
+     */
+    private static final DisplayImageOptions DISPLAY_IMAGE_OPTIONS
+            = new DisplayImageOptions.Builder().cacheInMemory(true).build();
+
+    private final ImageLoader imageLoader = ImageLoader.getInstance();
+
+    private OnSelectedListener onSelectedListener = null;
+    private OnCancelListener onCancelListener = null;
+
+    private ProgressBar progressBar;
+    private TextView emptyView;
+    private RecyclerView recyclerView;
+    private Disposable playlistsSubscriber;
+
+    private List<PlaylistLocalItem> playlists = new Vector<>();
+
+    public void setOnSelectedListener(final OnSelectedListener listener) {
+        onSelectedListener = listener;
+    }
+
+    public void setOnCancelListener(final OnCancelListener listener) {
```

Comment:
This doesn't seem to be called anywhere (same for the other Select*Fragments I think)

Reply:
Let's keep this here for now. Both settings and tabs need a full refactor anyway